docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc

```diff
@@ -7784,6 +7784,117 @@ SELECT (SELECT a FROM t1) FROM t GROUP BY (SELECT a FROM t1);
 SELECT a+1 FROM t GROUP BY 1+a;
 ```
 
+[[with_clause_syntax]]
+=== Syntax Description of WITH clause
+
+WITH clause, known as Common Table Expressions (CTEs) or subquery factoring clause, was introduced in the SQL-99 standard and has been implemented into Trafodion R2.1.
+
+For the complex expression which is referenced multiple times within the body of SELECT Statement, WITH clause assigns it an alias which is treated like a temporary table or an inline view that lasts only for the duration of the query and can be referenced various times in the same query.

```

Comment:
Minor grammar suggestion: Consider "For a complex expression which is referenced multiple times within the body of a SELECT statement, the WITH clause assigns it an alias. The alias is treated like a temporary table or an inline view..."  (I changed "the" to "a" before "complex expression, added "a" before SELECT, and made "statement" lower case.)